data_model/src/events/data/filters.rs

```diff
@@ -94,50 +94,60 @@ impl Filter for EntityFilter {
     }
 }
 
-#[derive(
-    Clone,
-    PartialOrd,
-    Ord,
-    PartialEq,
-    Eq,
-    Debug,
-    Decode,
-    Encode,
-    Deserialize,
-    Serialize,
-    IntoSchema,
-    Hash,
-)]
-/// Filter for identifiers
-///
-/// Passes id trough filter, if it equals to the one provided in construction
-pub struct IdFilter<Id: Eq>(Id);
+#[derive(Clone, PartialOrd, Ord, Eq, Debug, Decode, Encode, Serialize, IntoSchema)]
+/// Filter that accepts an data event whose origin matches what this filter specifies.
+pub struct OriginFilter<T: Origin>(<T::Origin as Identifiable>::Id);
 
-impl<Id: Eq> IdFilter<Id> {
-    /// Construct new `IdFilter`
-    pub fn new(id: Id) -> Self {
-        Self(id)
+impl<T: Origin> OriginFilter<T> {
+    /// Construct [`OriginFilter`].
+    pub fn new(origin_id: <T::Origin as Identifiable>::Id) -> Self {
+        Self(origin_id)
     }
 
-    /// Get `id`
-    pub fn id(&self) -> &Id {
+    /// Get the id of the origin of the data event that this filter accepts.
+    pub fn origin_id(&self) -> &<T::Origin as Identifiable>::Id {
         &self.0
     }
 }
 
-impl<Id: Eq> Filter for IdFilter<Id> {
-    type EventType = Id;
+impl<T: Origin> Filter for OriginFilter<T> {
+    type EventType = T;
+
+    fn matches(&self, event: &T) -> bool {
+        event.origin_id() == self.origin_id()
+    }
+}
+
+impl<T: Origin> PartialEq for OriginFilter<T> {
+    fn eq(&self, other: &Self) -> bool {
+        self.0 == other.0
+    }
+}
+
+impl<T: Origin> core::hash::Hash for OriginFilter<T> {
+    fn hash<H: core::hash::Hasher>(&self, state: &mut H) {
+        self.0.hash(state);
+    }
+}
```

Comment:
These implementations should be consistent each other. See [derive_hash_xor_eq](https://rust-lang.github.io/rust-clippy/master/index.html#derive_hash_xor_eq)